rosidl_generator_cpp/CMakeLists.txt

```diff
@@ -11,12 +11,56 @@ ament_export_include_directories(include)
 ament_python_install_package(${PROJECT_NAME})
 
 if(AMENT_ENABLE_TESTING)
-  find_package(ament_lint_auto REQUIRED)
-  ament_lint_auto_find_test_dependencies()
+
+  find_package(ament_cmake_nose REQUIRED)
+  find_package(rosidl_cmake REQUIRED)
+  find_package(rosidl_generator_c REQUIRED)
+
+  set(message_files
+    "msg/DynamicArrayNested.msg"
+    "msg/DynamicArrayPrimitives.msg"
+    "msg/Empty.msg"
+    "msg/Nested.msg"
+    "msg/NestedBounded.msg"
+    "msg/Primitives.msg"
+    "msg/PrimitivesBounded.msg"
+    "msg/StaticArrayNested.msg"
+    "msg/StaticArrayNestedBounded.msg"
+    "msg/StaticArrayPrimitives.msg"
+    "msg/StaticArrayBounded.msg"
+  )
+
+  set(service_files
+    "srv/Empty.srv"
+    "srv/PrimitivesBounded.srv"
+    "srv/DynamicArray.srv"
+  )
```

Comment:
I would consider removing the services since there i nothing different between them (requests / responses) and messages.